flight/PiOS/posix/pios_sys.c

```diff
@@ -232,6 +239,8 @@ static int handle_serial_device(const char *optarg) {
 		goto fail;
 	}
 
```

Comment:
This shouldn't be here, because of https://github.com/d-ronin/dRonin/blob/next/flight/Modules/GPS/GPS.c#L322
https://github.com/d-ronin/dRonin/blob/next/flight/Modules/Telemetry/telemetry.c#L903
etc

Reply:
Okay, I missed that, but it doesn't seem to be calling it correctly, as I see no evidence it has called PIOS_SERIAL_ChangeBaud in ...\PiOS\posix\pios_serial.c.  The GPS module does make the call correctly.  Confused......

Reply:
Could my issues/confusion stem from the fact I've got GPS hooked up thru the serial port, and the rf telemetry hooked up via the USB port, ttyAMA0 and ttyUSB0 respectively?

Reply:
It should all still work (though the TIOCSERIAL baud rate thing for the USB device is an issue, still).

Reply:
Okay, the main issue I was experiencing is beginning to become clearer, I think.  Without the changes in lines 227 and 242, there is no call made to PIOS_SERIAL_ChangeBaud by the telemetry subsystem.  The com_driver for the posix system is setup in line 197, and I believe the telemetry module is already initialized.   GPS works correctly because the module is enabled in line 214, so PIOS_SERIAL_ChangeBaud is correctly called during GPS init and config.

Reply:
I still am not able to debug, but that still seems strange.

In flightd/fw/main.c initTask SYS_Init and SYS_Args are called before MODULE_INITIALISE_ALL and MODULE_TASKCREATE_ALL.

Ah, I think I see it.  Will open a PR for discussion.

Reply:
Does #2085 help / seem appropriate?

Reply:
Will review it later, and test it this evening.